### PR TITLE
Timecourse regression model

### PR DESCRIPTION
**Pull request by namsaraeva**
A draft of the simple VGG-based neural network with a single output instead of the last classification layer.

### CODE REVIEW

File: src/sparcscore/ml/models.py [hunk 1]
```diff
@@ -158,6 +161,41 @@ def vgg(cfg, in_channels,  **kwargs):
         model = VGG2(self.make_layers(self.cfgs[cfg], in_channels), **kwargs)
         return model
 
+class VGG2_single_output(VGGBase):
```
Comment (sophiamaedler) [inline src/sparcscore/ml/models.py]: maybe better name VGG2_regression since you can also make single_output with the other implementation but this one does not have the softmax

**Comment by namsaraeva** [comment]
Note to self: look at AdamW as optimizer

**Review by sophiamaedler — COMMENTED** [review]
Looks good so far. Probably it would be a good idea to also implement AdamW as a potential optimiser.

**Comment by namsaraeva** [comment]
AdamW's weight decay default value is 1e-2, but I am not sure if I should add it as `self.hparams["weight_decay"] = 10 ** -2` too

File: src/sparcscore/ml/plmodels.py [hunk 2]
```diff
@@ -192,6 +192,9 @@ def configure_optimizers(self):
                 self.hparams["weight_decay"] = 0
             optimizer = torch.optim.Adam(self.parameters(), lr=self.hparams["learning_rate"], weight_decay=self.hparams["weight_decay"])
         elif self.hparams["optimizer"] == "AdamW":
+            #set weight decay to 1e-2 if not specified in hparams 
+            if self.hparams["weight_decay"] is None:
+                self.hparams["weight_decay"] = 10 ** -2
```
Comment (sophiamaedler) [inline src/sparcscore/ml/plmodels.py]: I think it's good to add it. then we ensure that we write out all required hparams

**Comment by namsaraeva** [comment]
It should be possible to pass on a column name instead of column index for target_col (the col with float timepoint targets) in HDF5SingleCellDatasetRegression

**Comment by namsaraeva** [comment]
@sophiamaedler should we stick to the VGG architecture with 11 or 16 layers? for MLP: config A or B? 

UPD: stick to the default architecture for now

**Comment by namsaraeva** [comment]
Consider adding HuberLoss instead of MSE, MAE

**Comment by namsaraeva** [comment]
Consider adding cross validation or grid search to test different settings for delta in Huber loss to get an optimal value.

"Huber loss will clip gradients to delta for residual (abs) values larger than delta. You want that when some part of your data points poorly fit the model and you would like to limit their influence."  - we probably want to limit the influence of phenotypes that are similar to previously seen phenotypes? @sophiamaedler

**Comment by namsaraeva** [comment]
Current limitation: scatterplot lims when training. We have to somehow pass the min and max values of dataset to the Callback module. 

Also implement a flexible Callback module that can be re-used.
